src/metric.cc

```diff
@@ -281,6 +362,13 @@ Metric::SetValue(const double& value)
   THROW_IF_TRITON_ERROR(TRITONSERVER_MetricSet(triton_metric, value));
 }
 
+void
+Metric::Observe(const double& value)
```

Comment:
Why not just use `Set`?

Reply:
In Prometheus, gauge calls `Set` to set itself to a value, while histogram and summary kinds call `Observe` to record a value internally. I think `Set` and `Observe` have different meanings although they are the same API. It also makes the call more clear to people who's familiar with Prometheus metrics. Open to suggestions.
https://github.com/jupp0r/prometheus-cpp/blob/master/core/include/prometheus/histogram.h#L58
https://github.com/jupp0r/prometheus-cpp/blob/master/core/include/prometheus/summary.h#L84

Reply:
Why we don't do it in core then?

Reply:
This is an internal implementation, let's keep it consistent with what we have in core.

Reply:
Agree, please keep in sync with the core API.

Reply:
Resolved.